### ISSUE TITLE
The problem of restarting the thread

### ISSUE THREAD

**Issue by lisicn** [justengel/pyjoystick#3]
Sir, I initiate an http request in django to start a thread to get the value of the handle, and then close it, this process is no problem. When I re-requested to start, it told me KeyError, I checked the values at this time: key, value, joystick, there is no difference from the first time I started, do you know why this is？
My environment：
os：windows10
python:3.8
django:3.1.5

http_app.views.py:
```python
...
flag = False
from http_app.utils.test import start_joystick

task_list = []

class StartJoystock(View):
    def get(self, request, *args, **kwargs):
        global flag
        if not flag:
            t = threading.Thread(target=start_joystick)
            t.start()
            task_list.append(t)
            message = "启动"
            flag = True

        else:
            message = "无法启动"
        return JsonResponse({"message": message})


class StopJoystock(View):
    def get(self, request, *args, **kwargs):

        global flag
        if flag:
            flag = False
            message = "关闭"
            t = task_list.pop()
            t.join(0)
        else:
            message = "无法关闭"

        return JsonResponse({"message": message})
...
```
http_app.utils.test.py:
```python
import time

from http_app.utils.connect_joystick import ConnectJoystack
from http_app.utils.serial_send import SendData


def start_joystick():
    '''
    com3 , 9600
    :return:
    '''
    com = SendData("com3", 9600)
    a = ConnectJoystack()
    a.mngr.start()
    while True:
        time.sleep(0.3)
        from http_app.views import flag
        if not flag:
            a.close()
            break
        com.data_send(a.direction, 152)
```
http_app.utils.connect_joystick.py:
```python
import pyjoystick
from pyjoystick.sdl2 import run_event_loop,rescale

'''
最终测试通过版本（手柄值初始化，发送至com口）
'''

class ConnectJoystack():
    def __init__(self):
        self.direction = 152
        self.speed = 152
        repeater = pyjoystick.HatRepeater(first_repeat_timeout=0.5, repeat_timeout=0.03, check_timeout=0.01)
        self.mngr = pyjoystick.ThreadEventManager(event_loop=run_event_loop,
                                                  handle_key_event=self.handle_key_event,
                                                  button_repeater=repeater)

    def handle_key_event(self,key):
        if key is None:
            # direction, speed = 152,152
            pass

        elif key.number == 0:  # 油门
            value = rescale(key.value, -1, 1, 100, 200)
            self.speed = int(value + 2)

        elif key.number == 3:  # 舵角
            value = rescale(key.value, -1, 1, 100, 200)
            self.direction = int(value + 2)

    def close(self):
        self.mngr.stop()
```
img:

![03](https://user-images.githubusercontent.com/38556416/108823485-711e8a00-75fb-11eb-8ae5-45a0ac832652.png)
success:
![01](https://user-images.githubusercontent.com/38556416/108823553-8dbac200-75fb-11eb-8366-72fb221197f1.png)
error:
![02](https://user-images.githubusercontent.com/38556416/108823566-927f7600-75fb-11eb-9856-5f0663d7b78c.png)

**Comment by justengel** [comment]
I designed this library thinking it would run the entire time the application is running. This library runs from events that the SDL library produces. In your case, you are getting the new joystick event. I use this to populate a dictionary to keep track of joysticks. You then stop using the ThreadEventManager and you start a new one. The new thread manager does not have knowledge of the joystick without receiving the new joystick event. 

This should be fixed in version 1.1.11

**Comment by lisicn** [comment]
sir,I understand, thanks for your reply
